### ISSUE TITLE
Manifest version update causes crashes when restoring with older version

### ISSUE THREAD

**Issue by kakoskin1** [duplicati/duplicati#2065]
I have:
- [x] searched open and closed issues for duplicates

---
### Version info

**Duplicati Version:** Backing up with 2.0.1.29_canary_2016-10-27, Restoring with 2.0.1.22_experimental_2016-09-12
**Operating System:** Fedora 24
**Backend:** File
### Bug description

Using an old version of Duplicati to restore a backup created with a newer version with incompatible manifest results in a crash. 
### Steps to reproduce
- (With version 2.0.1.29_canary_2016-10-27)
  duplicati-cli backup restoretest/ ~/git/ --no-encryption=1 --debug-output=1 --dbpath=crash-bu.sqlite

(with version 2.0.1.22_experimental_2016-09-12)
mono Duplicati-experimental/Duplicati.CommandLine.exe restore ~/temp/restoretest/ --no-encryption=1 --dbpath=crash.sqlite --debug-output=1 --log-file=duplicati-crash.log --log-level=Profiling --restore-path=restore

**Actual result:** Restore started at 30.10.2016 14:31:42
No local database, building a temporary database
  Listing remote folder ...
Rebuild database started, downloading 1 filelists
  Downloading file (126,55 KB) ...
Failed to process file: duplicati-20161030T120804Z.dlist.zip => Duplicati.Library.Main.Volumes.InvalidManifestException: Invalid manifest detected, the field Version has value 2 but the value 1 was expected 
at Duplicati.Library.Main.Volumes.VolumeReaderBase.UpdateOptionsFromManifest (System.String compressor, System.String file, Duplicati.Library.Main.Options options) <0x40284f20 + 0x001b3> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RecreateDatabaseHandler.DoRun (Duplicati.Library.Main.Database.LocalDatabase dbparent, Boolean updating, IFilter filter, Duplicati.Library.Main.Operation.NumberedFilterFilelistDelegate filelistfilter, Duplicati.Library.Main.Operation.BlockVolumePostProcessor blockprocessor) <0x4025d000 + 0x01bff> in <filename unknown>:0 
----Cut here for readability. See below for full backtrace.-----

**Expected result:** Warning saying something like:
This backup was created with newer incompatible version of Duplicati . 
You should use latest version of Duplicati to restore this backup.
### Full command output

Restore started at 30.10.2016 14:31:42
No local database, building a temporary database
  Listing remote folder ...
Rebuild database started, downloading 1 filelists
  Downloading file (126,55 KB) ...
Failed to process file: duplicati-20161030T120804Z.dlist.zip => Duplicati.Library.Main.Volumes.InvalidManifestException: Invalid manifest detected, the field Version has value 2 but the value 1 was expected
  at Duplicati.Library.Main.Volumes.VolumeReaderBase.UpdateOptionsFromManifest (System.String compressor, System.String file, Duplicati.Library.Main.Options options) <0x40284f20 + 0x001b3> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RecreateDatabaseHandler.DoRun (Duplicati.Library.Main.Database.LocalDatabase dbparent, Boolean updating, IFilter filter, Duplicati.Library.Main.Operation.NumberedFilterFilelistDelegate filelistfilter, Duplicati.Library.Main.Operation.BlockVolumePostProcessor blockprocessor) <0x4025d000 + 0x01bff> in <filename unknown>:0 
Filelists restored, downloading 1 index files
  Downloading file (62,63 KB) ...
Failed to process index file: duplicati-i83d904b6c2dd489194e63ba02aa5c00a.dindex.zip => Duplicati.Library.Main.Volumes.InvalidManifestException: Invalid manifest detected, the field Version has value 2 but the value 1 was expected
  at Duplicati.Library.Main.Volumes.VolumeBase+ManifestData.VerifyManifest (System.String manifest, Int64 blocksize, System.String blockhash, System.String filehash) <0x402a9830 + 0x002af> in <filename unknown>:0 
  at Duplicati.Library.Main.Volumes.VolumeReaderBase..ctor (ICompression compression, Duplicati.Library.Main.Options options) <0x402a8bc0 + 0x0018b> in <filename unknown>:0 
  at Duplicati.Library.Main.Volumes.VolumeReaderBase..ctor (System.String compressor, System.String file, Duplicati.Library.Main.Options options) <0x402a8a10 + 0x0003f> in <filename unknown>:0 
  at Duplicati.Library.Main.Volumes.IndexVolumeReader..ctor (System.String compressor, System.String file, Duplicati.Library.Main.Options options, Int64 hashsize) <0x402a89c0 + 0x00023> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RecreateDatabaseHandler.DoRun (Duplicati.Library.Main.Database.LocalDatabase dbparent, Boolean updating, IFilter filter, Duplicati.Library.Main.Operation.NumberedFilterFilelistDelegate filelistfilter, Duplicati.Library.Main.Operation.BlockVolumePostProcessor blockprocessor) <0x4025d000 + 0x0340f> in <filename unknown>:0 
Recreate completed, verifying the database consistency
Recreate completed, and consistency checks completed, marking database as complete
Checking remote backup ...
  Listing remote folder ...

System.Exception: No backup at the specified date
  at Duplicati.Library.Main.Database.LocalDatabase.GetFilesetIDs (DateTime restoretime, System.Int64[] versions) <0x402bf630 + 0x00333> in <filename unknown>:0 
  at Duplicati.Library.Main.Database.LocalRestoreDatabase.PrepareRestoreFilelist (DateTime restoretime, System.Int64[] versions, IFilter filter, ILogWriter log) <0x402bde60 + 0x001d3> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RestoreHandler.PrepareBlockAndFileList (Duplicati.Library.Main.Database.LocalRestoreDatabase database, Duplicati.Library.Main.Options options, IFilter filter, Duplicati.Library.Main.RestoreResults result) <0x402bd970 + 0x000cb> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RestoreHandler.DoRun (Duplicati.Library.Main.Database.LocalDatabase dbparent, IFilter filter, Duplicati.Library.Main.RestoreResults result) <0x402b60b0 + 0x0061b> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RestoreHandler.Run (System.String[] paths, IFilter filter) <0x4022b5f0 + 0x00a67> in <filename unknown>:0 
  at Duplicati.Library.Main.Controller+<Restore>c__AnonStorey1.<>m__0 (Duplicati.Library.Main.RestoreResults result) <0x40229e90 + 0x0007b> in <filename unknown>:0 
  at Duplicati.Library.Main.Controller.RunAction[T](Duplicati.Library.Main.T result, System.String[]& paths, IFilter& filter, System.Action`1 method) <0x4017b370 + 0x001c9> in <filename unknown>:0 
##

**Comment by agrajaghh** [comment]
why would you use an older version to restore?

**Comment by kakoskin1** [comment]
Primary reason to use older versions is restoring on another computer that happens to have older version installed. This would come up ,for example, after complete disk failure.

**Comment by kenkendk** [comment]
For this particular change, it is not possible to use an older version to restore.
The file format has changed slightly to fix some issues related to storing files with large chunks of metadata. The old version simply cannot parse the file-format, so the crash will happen.

I think the error message is pretty clear `Invalid manifest detected, the field Version has value 2 but the value 1 was expected`.

Do you want a message like: "You are attempting to restore from data created with version 2, but this install only supports version 1 files" ?

**Comment by agrajaghh** [comment]
maybe something like:

> You are attempting to restore from data created with a newer version of duplicati. Please update duplicati to successfully restore the files. Your backup data is using file version 2 but this install only supports version 1.

**Comment by kakoskin1** [comment]
Yes, that would be much better than current technical error message that is buried in the beginning of the backtrace. Backtraces are very unfriendly error messages to read unless you happen to be a programmer.
